cpp/arcticdb/version/version_map.hpp

```diff
@@ -254,13 +254,19 @@ class VersionMapImpl {
     std::pair<VersionId, std::vector<AtomKey>> tombstone_from_key_or_all(
             std::shared_ptr<Store> store,
             const StreamId& stream_id,
-            std::optional<AtomKey> first_key_to_tombstone = std::nullopt
+            std::optional<AtomKey> first_key_to_tombstone = std::nullopt,
+            std::optional<std::shared_ptr<VersionMapEntry>> cached_entry = std::nullopt
             ) {
-        auto entry = check_reload(
-            store,
-            stream_id,
-            LoadStrategy{LoadType::ALL, LoadObjective::UNDELETED_ONLY},
-            __FUNCTION__);
+        std::shared_ptr<VersionMapEntry> entry;
+        if (cached_entry)
```

Comment:
I'd rather not have the one line if as it's quite error prone in the face of merges / refactoring. Especially since here the statement in the `else` case spans multiple lines in the file.